### ISSUE TITLE
couldn't find file 'js-routes' with type 'application/javascript'...?

### ISSUE THREAD

**Issue by trlorenz** [railsware/js-routes#230]
[sprockets (3.7.1); sprockets-rails (3.2.1)]

Had occasion today to upgrade js-routes to the latest (1.4.1 -- we've been locked to 1.2.4 for a while), but had some trouble. Every request blows up with...

Sprockets::FileNotFound in SomeController#some_action, couldn't find file 'js-routes' with type 'application/javascript' Checked in these paths: [...]

...where nothing js-routes-ish occurs among the checked paths.

Traced the issue to 1.2.8; 1.2.7 seems to work fine for us; and in fact, 1.4.1 works for us, too, using 1.2.7's engine.rb.

I don't know much about Rails engines, and don't have time to dig into it right this second, but did discover that changing 1.4.1's 'Engine' class declaration...

```ruby
class Engine < ::Rails::Engine
  ...
end
```

...to be namespaced like so...

```ruby
class JsRoutes
  class Engine < ::Rails::Engine
    ...
  end
end
```

...seems to fix things. (The actual namespace doesn't seem to matter -- 'module WTF' works as well as 'class JsRoutes'.)

Anyone have a sense of what might be going on, here?

Thanks!

**Comment by bogdan** [comment]
It seems like a typo issue that `Engine` wasn't in the `JsRoutes` namespace. It've put it there and released 1.4.2. Please confirm that this version does work for you.

**Comment by trlorenz** [comment]
Yes, it does. Many thanks.
